app/code/Bulbulatory/Recomendations/view/adminhtml/ui_component/bulbulatory_recomendations_listing.xml

```diff
@@ -0,0 +1,106 @@
+<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
+    <argument name="data" xsi:type="array">
+        <item name="js_config" xsi:type="array">
+            <item name="provider" xsi:type="string">bulbulatory_recomendations_listing.bulbulatory_recomendations_listing_data_source</item>
+            <item name="deps" xsi:type="string">bulbulatory_recomendations_listing.bulbulatory_recomendations_listing_data_source</item>
+        </item>
+        <item name="spinner" xsi:type="string">spinner_columns</item>
+    </argument>
+    <dataSource name="bulbulatory_recomendations_listing_data_source">
+        <argument name="dataProvider" xsi:type="configurableObject">
+            <argument name="class" xsi:type="string">Magento\Framework\View\Element\UiComponent\DataProvider\DataProvider</argument>
+            <argument name="name" xsi:type="string">bulbulatory_recomendations_listing_data_source</argument>
+            <argument name="primaryFieldName" xsi:type="string">recommendation_id</argument>
+            <argument name="requestFieldName" xsi:type="string">recommendation_id</argument>
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="component" xsi:type="string">Magento_Ui/js/grid/provider</item>
+                    <item name="update_url" xsi:type="url" path="mui/index/render"/>
+                    <item name="storageConfig" xsi:type="array">
+                        <item name="indexField" xsi:type="string">recommendation_id</item>
+                    </item>
+                </item>
+            </argument>
+        </argument>
+    </dataSource>
+    <columns name="spinner_columns">
+        <column name="recommendation_id">
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="filter" xsi:type="string">textRange</item>
+                    <item name="label" xsi:type="string" translate="true">ID</item>
+                </item>
+            </argument>
+        </column>
+        <column name="recommender_id" class="Bulbulatory\Recomendations\Ui\Component\Listing\Column\RecommenderEmail">
```

Comment:
Pole nazywa się `recommender_id`, ale wyświetlany jest tam email - trochę to mylące. Skoro jest tutaj email, to najlepiej byłoby nazwać tę kolumnę np. `recommender_email`. Wtedy trzeba też zrobić zmianę w tym miejscu: https://github.com/maartynaa/Academy/pull/2/files#diff-592798128860763bef7da7c480f92ee8R31 - zamiast przypisywać do klucza `recommender_id`, trzeba będzie przypisać do klucza `recommender_email` (tak, aby klucz w tablicy `$item` odpowiadał nazwie kolumny w tym pliku).